python/ray/experimental/dag/class_node.py

```diff
@@ -63,57 +89,72 @@ def options(self, **options):
         return self
 
 
-class ClassMethodNode(DAGNode):
+class ClassMethodNode(ray_dag.DAGNode):
     """Represents an actor method invocation in a Ray function DAG."""
 
     def __init__(
         self,
-        actor: ClassNode,
-        prev_call: Optional["ClassMethodNode"],
-        method_name: str,
         method_args: Tuple[Any],
         method_kwargs: Dict[str, Any],
         method_options: Dict[str, Any],
+        kwargs_to_resolve: Dict[str, Any],
     ):
-        self._method_name: str = method_name
+        # TODO: (jiaodong) revisit constraints on dag INPUT before moving out
+        # of experimental folder
+        if ray_dag.INPUT in method_args and len(method_args) > 1:
```

Comment:
Can we make this a type that overrides `__reduce__` so it can't be accidentally captured even nested in args?

Reply:
good idea. clarifying question though -- won't we detect it in nested args due to the cloudpickling magic to find & replace nodes?

Reply:
We'd have to explicitly add a case looking for that in the cloudpickling scanner then. Right now it only intercepts DAGNode types.

Reply:
sg, this is more robust solution.